python/lsst/meas/algorithms/noise_covariance.py

```diff
@@ -0,0 +1,252 @@
+        """
+        bg = self.background.fitBackground(maskedImage)
+        bgImage = bg.getImageF(
+            self.background.config.algorithm, self.background.config.undersampleStyle
+        )
+        maskedImage -= bgImage
+        try:
+            yield
+        finally:
+            maskedImage += bgImage
+
+    def run(
+        self,
+        maskedImage: lsst.afw.image.MaskedImage,
+        refMaskedImage: lsst.afw.image.MaskedImage | None = None,
+    ) -> CorrelationMatrix:
+        """Compute the correlation matrix from a maskedImage.
+
+        Parameters
+        ----------
+        maskedImage :  `~lsst.afw.image.MaskedImage`
+            Image for which to determine the correlation matrix.
+        refMaskedImage : `~lsst.afw.image.MaskedImage`, optional
+            Image from which to determine which pixels to mask.
+            If None, it defaults to ``maskedImage``.
+
+        Returns
+        -------
+        corr_matrix : `CorrelationMatrix`
+            Correlation matrix of the maskedImage.
+
+        Raises
+        ------
+        RuntimeError
+            Raised if ``refMaskedImage`` is provided and does not have the same
+            dimensions as ``maskedImage``.
+        """
+        with self.subtractedBackground(maskedImage):
+            if refMaskedImage is None:
+                refMaskedImage = maskedImage
+            elif refMaskedImage.getDimensions() != maskedImage.getDimensions():
+                raise RuntimeError(
+                    "Reference image has different dimensions than input image"
+                )
+
+        corr_matrix = self._pixelBased(maskedImage, refMaskedImage)
+        return corr_matrix
+
+    def _pixelBased(
+        self,
+        maskedImage: lsst.afw.image.MaskedImage,
+        refMaskedImage: lsst.afw.image.MaskedImage,
+    ) -> CorrelationMatrix:
+        """Determine correlation coefficients between pixels
+
+        This is the concrete routine that does the computation.
+
+        Parameters
+        ----------
+        maskedImage : `~lsst.afw.image.MaskedImage`
+            Image for which to determine the variance rescaling factor.
+        refMaskedImage : `~lsst.afw.image.MaskedImage`
+            Image from which to determine which pixels to mask.
+
+        Returns
+        -------
+        corr_matrix : `CorrelationMatrix`
+            Correlation matrix of the maskedImage.
+        """
+        maskVal = refMaskedImage.mask.getPlaneBitMask(self.config.maskPlanes)
+        isGood = (
+            ((refMaskedImage.mask.array & maskVal) == 0)
+            & np.isfinite(refMaskedImage.image.array)
+            & np.isfinite(refMaskedImage.variance.array)
+            & (refMaskedImage.variance.array > 0)
+        )
+
+        nGood = np.sum(isGood)
+        self.log.debug(
+            "Number of selected background pixels: %d of %d.", nGood, isGood.size
+        )
+
+        normalized_arr = maskedImage.image.array / np.sqrt(maskedImage.variance.array)
```

Comment:
So I did,
```python
import numpy as np
np.ones((4,4))/np.zeros((4,4))
```
and did not get a DivideByZero error, but only a RuntimeWarning. The result was 4x4 matrix of `np.inf`, which I explicitly set `np.nan` in the very next line. So I don't think a try/except block is necessary here.

Reply:
I am capturing the warning so it doesn't confuse or raise a concern. And hiding the warning should be okay, since the zero division case is immediately handled by setting those pixel values to `NaN`s